paddle/fluid/extension/include/dispatch.h

```diff
@@ -28,19 +30,375 @@ namespace paddle {
 #define PD_PRIVATE_CASE_TYPE(NAME, enum_type, type, ...) \
   PD_PRIVATE_CASE_TYPE_USING_HINT(NAME, enum_type, type, data_t, __VA_ARGS__)
 
-#define PD_DISPATCH_FLOATING_TYPES(TYPE, NAME, ...)                          \
-  [&] {                                                                      \
-    const auto& dtype = TYPE;                                                \
-    switch (dtype) {                                                         \
-      PD_PRIVATE_CASE_TYPE(NAME, ::paddle::DataType::FLOAT32, float,         \
-                           __VA_ARGS__)                                      \
-      PD_PRIVATE_CASE_TYPE(NAME, ::paddle::DataType::FLOAT64, double,        \
-                           __VA_ARGS__)                                      \
-      default:                                                               \
-        throw std::runtime_error("function not implemented for this type."); \
-    }                                                                        \
+///////// Floating Dispatch Marco ///////////
+
+#define PD_DISPATCH_FLOATING_TYPES(TYPE, NAME, ...)                    \
```

Comment:
maybe permutation with num is better but i depends

Reply:
as discussed offline with @jzhang533 @saxon-zh , now we only add marco of specific data type, if needed, add other marco in the future